momento-sdk/src/main/java/momento/sdk/TopicClient.java

```diff
@@ -0,0 +1,104 @@
+package momento.sdk;
+
+import java.io.Closeable;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+import momento.sdk.auth.CredentialProvider;
+import momento.sdk.config.Configuration;
+import momento.sdk.responses.topic.TopicPublishResponse;
+import momento.sdk.responses.topic.TopicSubscribeResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Client to perform operations against the Momento Topic Service */
+public final class TopicClient implements Closeable {
+  private final ScsTopicClient scsTopicClient;
+
+  private final Logger logger = LoggerFactory.getLogger(TopicClient.class);
+
+  /**
+   * Constructs a TopicClient.
+   *
+   * @param credentialProvider Provider for the credentials required to connect to Momento.
+   * @param configuration Configuration object containing all tunable client settings.
+   */
+  public TopicClient(
+      @Nonnull CredentialProvider credentialProvider, @Nonnull Configuration configuration) {
+    this.scsTopicClient = new ScsTopicClient(credentialProvider, configuration);
+    logger.info("Creating Momento Topic Client");
+    logger.debug("Cache endpoint: " + credentialProvider.getCacheEndpoint());
+    logger.debug("Control endpoint: " + credentialProvider.getControlEndpoint());
+  }
+
+  /**
+   * Constructs a TopicClient.
+   *
+   * @param credentialProvider Provider for the credentials required to connect to Momento.
+   * @param configuration Configuration object containing all tunable client settings.
+   * @return TopicClient
+   */
+  public static TopicClient create(
+      @Nonnull CredentialProvider credentialProvider, @Nonnull Configuration configuration) {
```

Comment:
In our other sdks, we distinguish between `Configuration`, which is a cache configuration, and `TopicConfiguration` which are configurations specific to the topic client

Reply:
I will add TopicConfiguration to our configurations and use that for the topic clients. Thanks!